src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -75,30 +75,37 @@ function NewContactMethodPage(props) {
         return '';
     };
 
-    const validate = (values) => {
-        const phoneLogin = getPhoneLogin(values.phoneOrEmail);
-        const validateIfnumber = validateNumber(phoneLogin);
-
-        const errors = {};
-
-        if (_.isEmpty(values.phoneOrEmail)) {
-            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', 'contacts.genericFailureMessages.contactMethodRequired');
-        }
-
-        if (!_.isEmpty(values.phoneOrEmail) && !(parsePhoneNumber(phoneLogin).possible || Str.isValidEmail(values.phoneOrEmail))) {
-            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', 'contacts.genericFailureMessages.invalidContactMethod');
-        }
-
-        if (!_.isEmpty(values.phoneOrEmail) && lodashGet(props.loginList, validateIfnumber || values.phoneOrEmail.toLowerCase())) {
-            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', 'contacts.genericFailureMessages.enteredMethodIsAlreadySubmited');
-        }
-
-        if (!Permissions.canUsePasswordlessLogins(props.betas) && _.isEmpty(values.password)) {
-            errors.password = 'contacts.genericFailureMessages.passwordRequired';
-        }
-
-        return errors;
-    };
+    const validate = React.useCallback(
+        (values) => {
+            const phoneLogin = getPhoneLogin(values.phoneOrEmail);
+            const validateIfnumber = validateNumber(phoneLogin);
+
+            const errors = {};
+
+            if (_.isEmpty(values.phoneOrEmail)) {
+                ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', 'contacts.genericFailureMessages.contactMethodRequired');
+            }
+
+            if (!_.isEmpty(values.phoneOrEmail) && !(parsePhoneNumber(phoneLogin).possible || Str.isValidEmail(values.phoneOrEmail))) {
+                ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', 'contacts.genericFailureMessages.invalidContactMethod');
+            }
+
+            if (!_.isEmpty(values.phoneOrEmail) && lodashGet(props.loginList, validateIfnumber || values.phoneOrEmail.toLowerCase())) {
+                ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', 'contacts.genericFailureMessages.enteredMethodIsAlreadySubmited');
+            }
+
+            if (!Permissions.canUsePasswordlessLogins(props.betas) && _.isEmpty(values.password)) {
+                errors.password = 'contacts.genericFailureMessages.passwordRequired';
+            }
+
+            return errors;
+        },
+        // We don't need `props.loginList` because when submitting this form
+        // the props.loginList gets updated, causing this function to run again.
```

Comment:
Why does the `props.loginList` gets updated? is there a way to prevent this?

Reply:
I see.. 😬 

@allroundexperts can we move `validateNumber`, `getPhoneLogin` and `addNewContactMethod` out the component please, they are unnecessarily recreating on each render and the first two are used by `validate`